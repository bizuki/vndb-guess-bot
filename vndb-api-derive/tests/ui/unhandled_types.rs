use vndb_api_derive::VndbFieldsEnum;

#[derive(VndbFieldsEnum)]
pub struct Struct {
    str: &'static str,
    i32: &'static i32,
}

fn main() {
    let _ = StructFields::Str;
    let _ = StructFields::I32;
}
