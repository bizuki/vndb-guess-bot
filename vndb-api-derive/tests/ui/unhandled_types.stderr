error: #[derive(VndbFieldsEnum)] unsupported field type
 --> tests/ui/unhandled_types.rs:5:10
  |
5 |     str: &'static str,
  |          ^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `StructFields`
  --> tests/ui/unhandled_types.rs:10:13
   |
10 |     let _ = StructFields::Str;
   |             ^^^^^^^^^^^^ use of undeclared type `StructFields`

error[E0433]: failed to resolve: use of undeclared type `StructFields`
  --> tests/ui/unhandled_types.rs:11:13
   |
11 |     let _ = StructFields::I32;
   |             ^^^^^^^^^^^^ use of undeclared type `StructFields`
